tests/manage/pv_services/pvc_snapshot/test_pvc_snapshot.py

```diff
@@ -0,0 +1,175 @@
+        assert err_count == 0, (
+            f"IO error on pod {self.pod_obj.name}. "
+            f"FIO result: {fio_result}"
+        )
+        log.info(f"Verified IO on pod {self.pod_obj.name}.")
+
+        # Verfiy presence of the file
+        file_path = pod.get_file_path(self.pod_obj, file_name)
+        log.info(file_path)
+        assert pod.check_file_existence(self.pod_obj, file_path), (
+            f"File {file_name} doesn't exist"
+        )
+        log.info(f"File {file_name} exists in {self.pod_obj.name}")
+
+        # Calculate md5sum
+        orig_md5_sum = pod.cal_md5sum(self.pod_obj, file_name)
+        # Take a snapshot
+        snap_yaml = constants.CSI_RBD_SNAPSHOT_YAML
+        snap_name = helpers.create_unique_resource_name(
+            'test', 'snapshot'
+        )
+        snap_obj = self.pvc_obj.create_pvc_snapshot(
+            snap_yaml,
+            snap_name,
+            self.snapshot_sc_obj.name,
+        )
+        snap_obj.ocp.wait_for_resource(
+            condition='true', resource_name=snap_obj.name,
+            column=constants.STATUS_READYTOUSE, timeout=60
+        )
+
+        # Same Storage class of the original PVC
+        sc_name = self.pvc_obj.backed_sc
+
+        # Size should be same as of the original PVC
+        pvc_size = str(self.pvc_obj.size) + "Gi"
+
+        # Create pvc out of the snapshot
+        # Both, the snapshot and the restore PVC should be in same namespace
+        restore_pvc_obj = helpers.create_restore_pvc(
+            sc_name=sc_name, snap_name=snap_obj.name,
+            namespace=snap_obj.namespace, size=pvc_size
+        )
+        helpers.wait_for_resource_state(
+            restore_pvc_obj,
+            constants.STATUS_BOUND
+        )
+        restore_pvc_obj.reload()
+
+        # Create and attach pod to the pvc
+        restore_pod_obj = pod_factory(
+            interface=interface,
+            pvc=restore_pvc_obj,
+            status=constants.STATUS_RUNNING
+        )
+
+        # Verify that the file is present on the new pod
+        log.info(
+            f"Checking the existence of {file_name} "
+            f"on restore pod {restore_pod_obj.name}"
+        )
+        assert pod.check_file_existence(restore_pod_obj, file_path), (
+            f"File {file_name} doesn't exist"
+        )
+        log.info(f"File {file_name} exists in {restore_pod_obj.name}")
+
+        # Verify that the md5sum matches
+        log.info(
+            f"Verifying that md5sum of {file_name} "
+            f"on pod {self.pod_obj.name} matches with md5sum "
+            f"of the same file on restore pod {restore_pod_obj.name}"
+        )
+        assert pod.verify_data_integrity(
+            restore_pod_obj,
+            file_name,
+            orig_md5_sum
+        ), 'Data integrity check failed'
+        log.info(f'{"Data integrity check passed, md5sum are same"}')
+
+        # Delete the snapshot
+        teardown_factory(snap_obj)
```

Comment:
Please call this right after the creation of each resource.
This is important in case the test will fail at some point and when those lines are last, the test will leave leftovers

Reply:
Sure Raz, have made the changes. Want to know if I force push in the same commit or create a different commit for review comments?